plugins/document/document.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2021 Nitric Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package document
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/nitric-dev/membrane/sdk"
+	"github.com/nitric-dev/membrane/utils"
+)
+
+const ATTRIB_PK = "_pk"
+
+const ATTRIB_SK = "_sk"
+
+const ROOT_SK = "ROOT#"
+
+// Map of valid expression operators
+var validOperators = map[string]bool{
+	"==":         true,
+	">":          true,
+	"<":          true,
+	">=":         true,
+	"<=":         true,
+	"startsWith": true,
+}
+
+var stack *utils.NitricStack
+
+func Stack() (*utils.NitricStack, error) {
+	if stack == nil {
+		nitricStack, err := utils.NewStackDefault()
+		if err != nil {
+			return nil, fmt.Errorf("error loading Nitric stack definition: %v", err)
+		}
+		stack = nitricStack
+	}
+	return stack, nil
+}
```

Comment:
As discussed we won't need the definition for now, should probably be in a different package as well.